pkg/nuctl/command/deploy.go

```diff
@@ -638,6 +655,36 @@ func (d *deployCommandeer) enrichConfigWithComplexArgs() error {
 	return nil
 }
 
+func (d *deployCommandeer) enrichBuildConfigWithArgs() {
+
+	// enrich string fields in function config with flags
+	for flagValue, fieldInFunctionConfig := range map[string]*string{
+		d.functionConfigPath:               &d.functionConfig.Spec.Build.FunctionConfigPath,
+		d.functionBuild.Path:               &d.functionConfig.Spec.Build.Path,
+		d.functionBuild.FunctionSourceCode: &d.functionConfig.Spec.Build.FunctionSourceCode,
+		d.functionBuild.Image:              &d.functionConfig.Spec.Build.Image,
+		d.functionBuild.Registry:           &d.functionConfig.Spec.Build.Registry,
+		d.functionBuild.BaseImage:          &d.functionConfig.Spec.Build.BaseImage,
+		d.functionBuild.OnbuildImage:       &d.functionConfig.Spec.Build.OnbuildImage,
+		d.functionBuild.CodeEntryType:      &d.functionConfig.Spec.Build.CodeEntryType,
+	} {
+		if flagValue != "" {
+			*fieldInFunctionConfig = flagValue
+		}
+	}
+
+	// enrich bool fields in function config with flags
+	for flagValue, fieldInFunctionConfig := range map[bool]*bool{
+		d.functionBuild.NoBaseImagesPull: &d.functionConfig.Spec.Build.NoBaseImagesPull,
+		d.functionBuild.NoCleanup:        &d.functionConfig.Spec.Build.NoCleanup,
+		d.functionBuild.Offline:          &d.functionConfig.Spec.Build.Offline,
+	} {
+		if flagValue {
+			*fieldInFunctionConfig = flagValue
+		}
+	}
+}
```

Comment:
not a must, but to save some dups, you could probably implement this once as a go generic function that accepts a `map[T]*T`